ingestion-beam/src/main/java/com/mozilla/telemetry/heka/HekaTransform.java

```diff
@@ -0,0 +1,33 @@
+package com.mozilla.telemetry.heka;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.List;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.options.ValueProvider;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.values.PBegin;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubMessage;
+import org.apache.beam.sdk.io.FileIO.ReadableFile;
+import org.apache.beam.sdk.io.fs.MatchResult;
+import org.json.JSONObject;
+import com.mozilla.telemetry.transforms.WithErrors;
+import com.mozilla.telemetry.transforms.MapElementsWithErrors;
+
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.ParDo;
+
+public class HekaTransform extends MapElementsWithErrors<ReadableFile, List<PubsubMessage>> {
```

Comment:
Beam provides some naming guidelines for transforms: https://beam.apache.org/contribute/ptransform-style-guide/#naming

In that sense, the best name for this would probably be `HekaIO.read()`, which would mean the outer `Heka` class doesn't extend anything, but would have a single static `read()` method that returns an instance of `HekaIO.Read()` which would be the thing extending MapElementsWithErrors.

I don't think it's worth worrying about giving this a pretty name right now. I'm mostly just trying to continue giving context on Beam as we go here.